client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java

```diff
@@ -101,13 +102,30 @@ public KaaClientPropertiesState(PersistentStorage storage, Base64 base64, KaaCli
         super();
         this.storage = storage;
         this.base64 = base64;
-        stateFileLocation = properties.containsKey(STATE_FILE_LOCATION) ? properties.getProperty(STATE_FILE_LOCATION) : STATE_FILE_DEFAULT;
 
-        clientPrivateKeyFileLocation = properties.containsKey(CLIENT_PRIVATE_KEY_FILE_LOCATION) ? properties
-                .getProperty(CLIENT_PRIVATE_KEY_FILE_LOCATION) : CLIENT_PRIVATE_KEY_DEFAULT;
+        properties.setBase64(base64);
+
+        String workingDir = properties.getWorkingDirectory();
+        if (isBlank(workingDir)) {
```

Comment:
All this logic need to be integrated to KaaClientProperties in order to avoid code duplications in future